### PR TITLE
feat: bot service (AR-2011)

### PR DESCRIPTION
**Pull request by Garzas**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Adds bot service model in user to differentiate them from normal users. This will help in future for creating conversation with bots. For now it will just hide connection/open conversation button in bot user profile

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/publicuser/PublicUserMapper.kt [hunk 1]
```diff
@@ -74,7 +77,8 @@ class PublicUserMapperImpl(
         completePicture = userDetailResponse.assets.getCompleteAssetOrNull()
             ?.let { idMapper.toQualifiedAssetId(it.key, userDetailResponse.id.domain) },
         availabilityStatus = UserAvailabilityStatus.NONE,
-        userType = userType
+        userType = userType,
+        botService = userDetailResponse.service?.let { BotService(it.id, it.provider) }
```
Comment (yamilmedina) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/publicuser/PublicUserMapper.kt:81]: Just a question to understand more about the context. Does this data comes from this same endpoints as well ? 
And if true, that means Bots are like "users" but with some extra data? 🤔  Did I catch it right?

File: persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/BotServiceAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.wire.kalium.persistence.dao
+
+import app.cash.sqldelight.ColumnAdapter
+
+class BotServiceAdapter : ColumnAdapter<BotEntity, String> {
```
Comment (yamilmedina) [inline persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/BotServiceAdapter.kt:5]: Doesn't this fall into the same category of qualified IDs ? Just asking to see if there were some re-utilization case or not 😅 ?

Which values are possibles in the `provider` field ?

**Review by yamilmedina — APPROVED** [review]
lgtm! Just added some questions regarding some unknow concepts to be more clear. But anyway great job ! 🛴 🚀

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/publicuser/PublicUserMapper.kt (hunk 1 above)
Comment (Garzas) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/publicuser/PublicUserMapper.kt:81]: Yup, exactly

File: persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/BotServiceAdapter.kt (hunk 2 above)
Comment (Garzas) [reply persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/BotServiceAdapter.kt:5]: Currently we don't have provider name which probably will be added to bot service in future 🤔

**Pull request merged**